### ISSUE TITLE
[BUG 3.13] Observer on CP receiving wrong value

### ISSUE THREAD

**Issue by simonihmig** [emberjs/ember.js#18246]
Here is a reproduction, that is failing starting w/ 3.13.0-beta.1 (same on canary): https://codesandbox.io/embed/my-app-i4i02?initialpath=/tests

The failing test is this one:

```js
  test("case 3", function(assert) {
    assert.expect(3);

    const Dummy = EmberObject.extend({
      foo: true,

      _foo: computed("foo", {
        get() {
          return this.get("foo");
        },
        set(key, value) {
          // eslint-disable-line no-unused-vars
          return value;
        }
      }),

      _fooTracker: observer("_foo", function() {
        let value = this.get("_foo");
        assert.notOk(value);
      })
    });

    const subject = Dummy.create();

    assert.ok(subject.get("_foo"));
    subject.set("_foo", false);
    assert.notOk(subject.get("_foo"));
  });
```

`_foo` in the observer function is (still) `true` (makes the assertion fail), while it should be `false` as the observer is triggered by setting it explicitly to `false`. Another super weird thing is that the third assertion (the last line in the test) now also fails, while it does *not* fail when `_foo` is not read from in the observer (see the "case2" test in the sandbox).

Also the observer does see the updated value of `true`, when not observing the CP `_foo`, but directly the simple property `foo` (see "case1").

**Comment by pzuraq** [comment]
I believe this may be fixed by some changes in #18223

**Comment by rwjblue** [comment]
We should ensure (by adding a test)...

**Comment by simonihmig** [comment]
Ok, I'll try to add a PR with that failing test...

**Comment by simonihmig** [comment]
Here's a failing test: https://github.com/emberjs/ember.js/pull/18248

**Comment by simonihmig** [comment]
Fixed by #18248!
